src/sixty_four.rs

```diff
@@ -43,9 +43,9 @@ impl XxCore {
     }
 
     #[inline(always)]
-    fn ingest_chunks<'a, I>(&mut self, values: I)
+    fn ingest_chunks<I>(&mut self, values: &mut I)
     where
-        I: IntoIterator<Item = &'a [u64; 4]>,
+        I: Iterator<Item = [u64; 4]>,
```

Comment:
Can't this stay as `IntoIterator` and just `I`? We should be able to pass in `&mut whatever` still.

Reply:
In case the user provides a buffer with 32 + 3 bytes, it's one iteration of `[u64; 4]`. If the iterator (`UnalignedBuffer`) gets consumed here, the callsite can no longer read the remaining 3 bytes. Practically, this does not seem to hurt performance (I tested the type change with the old code).

Reply:
> If the iterator (`UnalignedBuffer`) gets consumed here

Right, the caller will still pass in `&mut whatever` — `Iterator` is implemented for `&mut impl Iterator`, and `IntoIterator` is implemented for `impl Iterator`.